### PR TITLE
[Merged by Bors] - feat(Analysis/BoxIntegral/Basic): a bounded, a.e. continuous function is integrable on a box

### PR DESCRIPTION
**Pull request by js2357**
Prove that a function that is bounded and a.e. continuous within a box is integrable on that box. This generalizes integrable_of_continuousOn.

As an intermediate step, define the oscillation of a function at a point x (and oscillationWithin for the oscillation at x within a specific subset). Also prove some lemmas about oscillation, namely that oscillation is 0 at a point if and only if the function is continuous there, and a statement about uniformity of oscillation on a compact set, with versions for oscillation and oscillationWithin.

This is essentially the same as #13013, but I'm making a new PR because I messed up the other branch.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by xroblot — APPROVED** [review]
LGTM! The new version is very nice :)

File: Mathlib/Analysis/BoxIntegral/Basic.lean [hunk 1]
```diff
@@ -673,42 +674,140 @@ open MeasureTheory
...
+     two tagged prepartitions π₁ and π₂ can be made ε-close by making the partitions
+     sufficiently fine.
+
+     Start by defining some constants C, ε₁, ε₂ that will be useful later. -/
+  refine integrable_iff_cauchy_basis.2 fun ε ε0 ↦ ?_
+  rcases exists_pos_mul_lt ε0 (2 * μ.toBoxAdditive I) with ⟨ε₁, ε₁0, hε₁⟩
+  rcases hb with ⟨C, hC⟩
+  by_cases C0 : C ≥ 0; swap
+  · obtain ⟨x, hx⟩ := BoxIntegral.Box.nonempty_coe I
+    exact False.elim <| C0 <| le_trans (norm_nonneg (f x)) <| hC x (I.coe_subset_Icc hx)
```
Comment (Ruben-VandeVelde) [inline Mathlib/Analysis/BoxIntegral/Basic.lean:696]: This is more straightforward as:
```suggestion
  have C0 : 0 ≤ C := by
    obtain ⟨x, hx⟩ := BoxIntegral.Box.nonempty_coe I
    exact le_trans (norm_nonneg (f x)) <| hC x (I.coe_subset_Icc hx)
```

File: Mathlib/Analysis/BoxIntegral/Basic.lean [hunk 2]
```diff
@@ -673,42 +674,140 @@ open MeasureTheory
...
+  rcases exists_pos_mul_lt ε0 (4 * C) with ⟨ε₂, ε₂0, hε₂⟩
+  have ε₂0': ENNReal.ofReal ε₂ ≠ 0 := fun h ↦ not_le_of_gt ε₂0 (ofReal_eq_zero.1 h)
```
Comment (Ruben-VandeVelde) [inline Mathlib/Analysis/BoxIntegral/Basic.lean:698]: Definitely add a space before the colon, and maybe also:
```suggestion
  have ε₂0' : ENNReal.ofReal ε₂ ≠ 0 := ne_of_gt <| ofReal_pos.2 ε₂0
```

**Review by Ruben-VandeVelde — APPROVED** [review]
Two minor comments; I trust @xroblot has reviewed the mathematics.

**Comment by xroblot** [comment]
> Two minor comments; I trust @xroblot has reviewed the mathematics.

Well, I am no an expert in this domain (although I know more and more about it thanks to Lean :wink:) but I can say two things: 1) the statement proved here in the same as the one on Wikipedia; 2) it fits perfectly with my application to number theory.

**Review by Ruben-VandeVelde — APPROVED** [review]
Thanks!

maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**